File: week-03/day-04/src/Power.java
public class Power {
    public static void main(String[] args) {

// Given base and n that are both 1 or more, compute recursively (no loops)
// the value of base to the n power, so powerN(3, 2) is 9 (3 squared).
        System.out.println(powerN(4, 1));

    }

    public static int powerN(int base, int power) {
        if (base == 1 || power == 1) {
            return base;
        }
        return base * powerN(base, power - 1);
    }
}

File: PythonPractice/0102/power.py

# 4. Given base and n that are both 1 or more, compute recursively (no loops)
# the value of base to the n power, so powerN(3, 2) is 9 (3 squared).


def powerN(base, power):
    if base == 1 or power == 1:
        return base
    else:
        print(base * powerN(base, power - 1))
        return base * powerN(base, power - 1)


print(powerN(2, 10))
